src/wallet/wallet.cpp

```diff
@@ -2969,7 +2969,9 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
             // Create change script that will be used if we need change
             // TODO: pass in scriptChange instead of reservedest so
             // change transaction isn't always pay-to-bitcoin-address
-            CScript scriptChange;
+            CScript scriptChange = CScript();
+            bool failed_to_get_change_address = false;
+            std::string failed_to_get_change_address_reason = "";
```

Comment:
1cb1381ca51d1967146a0a75fd96c1b4800ec378

I think you don't need this, you can write the reason in `strFailReason`.